src/System.Data.Common/src/System/Data/EnumerableRowCollection.GetLinqDataView.cs

```diff
@@ -0,0 +1,160 @@
+                    };
+            }
+            #endregion BuildSinglePredicate
+
+            #region Evaluate Filter/Sort
+            //  All of this mess below is because we want to create index only once.
+            //
+            //  If we only have filter, we set _view.Predicate       - 1 index creation
+            //  If we only have sort, we set _view.SortExpression()  - 1 index creation
+            //  If we have BOTH, we set them through the constructor - 1 index creation
+            //
+
+            // Filter AND Sort
+            if ((null != finalPredicate) && (0 < _sortExpression.Count))
+            {
+                // A lot more work here because constructor does not know type K,
+                // so the responsibility to create appropriate delegate comparers
+                // is outside of the constructor.
+
+                view = new LinqDataView(
+                               _table,
+                               finalPredicate,                      //Func() Predicate
+                               delegate(DataRow row)                //System.Predicate
+                               {
+                                   return finalPredicate(row);
+                               },
+                               delegate(DataRow a, DataRow b)       //Comparison for DV for Index creation
+                               {
+                                   return _sortExpression.Compare(
+                                            _sortExpression.Select((TRow)(object)a),
+                                            _sortExpression.Select((TRow)(object)b)
+                                       );
+                               },
+                               delegate(object key, DataRow row)    //Comparison_K_T for DV's Find()
+                               {
+                                   return _sortExpression.Compare(
+                                        (List<object>)key,
+                                        _sortExpression.Select((TRow)(object)row)
+                                      );
+                               },
+                                _sortExpression.CloneCast<DataRow>());
+            }
+            else if (null != finalPredicate)
+            {
+                //Only Filtering
+                view = new LinqDataView(
+                                    _table,
+                                    finalPredicate,
+                                    delegate(DataRow row)                //System.Predicate
+                                    {
+                                        return finalPredicate(row);
+                                    },
+                                    null,
+                                    null,
+                                    _sortExpression.CloneCast<DataRow>());
+            }
+            else if (0 < _sortExpression.Count)
+            {
+                //Only Sorting
```

Comment:
If you feel like it, it would be nice to regex a space after all `//` if missing.